code/model/modelanimation.cpp

```diff
@@ -788,6 +789,11 @@ namespace animation {
 			anim->m_instances[pmi->id].instance_flags.set(flag, set);
 	}
 
+	void ModelAnimationSet::AnimationList::setSpeed(float speed) const {
+		for (auto anim : animations)
+			anim->m_instances[pmi->id].speed = speed;
+	};
```

Comment:
Is this actually a `const` function if it mutates its own state?

And it looks like it doesn't actually mutate it, since the loop variable `anim` is a copy, not a reference.

Reply:
it creates a copy of a shared_ptr, of which it modifies the data.
Yes, this could also be a const reference to the shared_ptr (and should be, really). But since it just modifies the ptr's data and not the ptr itself, the List, which just holds the ptr stays constant

Reply:
Weird but ok.